src/conversions/length_conversion.rs

```diff
@@ -0,0 +1,107 @@
+/// Author : https://github.com/ali77gh
+/// Conversion of length units.
+///
+/// Available Units:
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Millimeter
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Centimeter
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Meter
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Kilometer
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Inch
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Foot
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Yard
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Mile
+
+/// Universal Units on Length
+pub enum LengthUnit {
+    Millimeter,
+    Centimeter,
+    Meter,
+    Kilometer,
+    Inch,
+    Foot,
+    Yard,
+    Mile,
+}
+
+/// Private methods
+/// This is a n*n problem
+/// It's gonna be 56 functions or 56 cases in match statement
+/// It's hard to write code for every unit to unit so I solve this problem
+/// by converting input to meter and than convert it to output
+impl LengthUnit {
+    /// This function give you a number (let's call it n)
+    /// So if you multiple a value in this unit to n you will get meters
+    ///
+    /// m * in-unit = meter
+    fn get_unit_to_meter_multiplier(&self) -> f64 {
+        match self {
+            LengthUnit::Millimeter => 0.001,
+            LengthUnit::Centimeter => 0.01,
+            LengthUnit::Meter => 1.0,
+            LengthUnit::Kilometer => 1000.0,
+            LengthUnit::Inch => 0.0254,
+            LengthUnit::Foot => 0.3048,
+            LengthUnit::Yard => 0.9144,
+            LengthUnit::Mile => 1609.34,
+        }
+    }
+
+    /// This function give you a number (let's call it n)
+    /// So if you multiple a value in meters to n you will get unit
+    ///
+    /// m * meter = in-unit
+    fn get_unit_from_meter_multiplier(&self) -> f64 {
+        1.0 / self.get_unit_to_meter_multiplier()
+    }
+}
```

Comment:
Could we just use non-public functions for that?

Reply:
Hmm🤔
I don't understand you. 
These functions are non-public (private) 
Isn't it?

If you mean I should make these public. the caller can easily do this:
```rust
length_conversion(value, LengthUnit::Meter, Unit);
// and reverse:
length_conversion(value, Unit, LengthUnit::Meter);
```

Should we have different API just for meter

Or maybe I should move my length_conversion to impl block?

Can you explain a bit more?

Reply:
The entire implementation could look like this:
```rust
fn unit_to_meter_multiplier(from: LengthUnit) -> f64 {
    match from {
        LengthUnit::Millimeter => 0.001,
        LengthUnit::Centimeter => 0.01,
        LengthUnit::Meter => 1.0,
        LengthUnit::Kilometer => 1000.0,
        LengthUnit::Inch => 0.0254,
        LengthUnit::Foot => 0.3048,
        LengthUnit::Yard => 0.9144,
        LengthUnit::Mile => 1609.34,
    }
}

fn to_meter(input: f64, from: LengthUnit) -> f64 {
    input * unit_to_meter_multiplier(from)
}

fn meter_to_unit(input: f64, to: LengthUnit) -> f64 {
    input / unit_to_meter_multiplier(to)
}

pub fn length_conversion(input: f64, from: LengthUnit, to: LengthUnit) -> f64 {
    meter_to_unit(to_meter(input, from), to)
}
```

In my opinion, it is easier to think about _conversion to meter_ and _conversion from meter_ than about multipliers.

Reply:
I agree. It's cleaner👍.
I did change it 
https://github.com/ali77gh/TheAlgorithmsRust/commit/ea60fed02ae820e8968b6c8ce8e7c205cd6f66b1.